pkg/informers/informers.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2020 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package informers
+
+import (
+	"strings"
+	"time"
+
+	"github.com/prometheus-operator/prometheus-operator/pkg/listwatch"
+
+	"github.com/pkg/errors"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+)
+
+type GenericInformer interface {
```

Comment:
`GenericInformer` is a bit misleading IMO, as the canonical definition of what comprises and informer is in the client-go codebase. Consider maybe `InformLister` following the example from the golang io package of composed informers of e.g. `ReadWriter`.

Reply:
nice, that's a much better and more idiomatic naming :+1:

Reply:
fixed